sklearn/decomposition/tests/test_pca.py

```diff
@@ -9,130 +10,55 @@
 from sklearn.utils.testing import assert_allclose
 from sklearn.utils.testing import assert_equal
 from sklearn.utils.testing import assert_greater
-from sklearn.utils.testing import assert_raise_message
-from sklearn.utils.testing import assert_raises
-from sklearn.utils.testing import assert_raises_regex
 from sklearn.utils.testing import assert_no_warnings
-from sklearn.utils.testing import ignore_warnings
-from sklearn.utils.testing import assert_less
 
 from sklearn import datasets
 from sklearn.decomposition import PCA
 from sklearn.decomposition.pca import _assess_dimension_
 from sklearn.decomposition.pca import _infer_dimension_
 
 iris = datasets.load_iris()
-solver_list = ['full', 'arpack', 'randomized', 'auto']
+PCA_SOLVERS = ['full', 'arpack', 'randomized', 'auto']
 
 
-def test_pca():
-    # PCA on dense arrays
+@pytest.mark.parametrize(
+    'svd_solver, n_components, err_msg',
+    [('arpack', 0, r'must be between 1 and min\(n_samples, n_features\)'),
+     ('randomized', 0, r'must be between 1 and min\(n_samples, n_features\)'),
+     ('arpack', iris.data.shape[1], r'must be strictly less than min')]
+)
+def test_pca_internal_state(svd_solver, n_components, err_msg):
     X = iris.data
+    pca_fitted = PCA(n_components=n_components, svd_solver=svd_solver)
 
-    for n_comp in np.arange(X.shape[1]):
-        pca = PCA(n_components=n_comp, svd_solver='full')
-
-        X_r = pca.fit(X).transform(X)
-        np.testing.assert_equal(X_r.shape[1], n_comp)
-
-        X_r2 = pca.fit_transform(X)
-        assert_array_almost_equal(X_r, X_r2)
-
-        X_r = pca.transform(X)
-        X_r2 = pca.fit_transform(X)
-        assert_array_almost_equal(X_r, X_r2)
-
-        # Test get_covariance and get_precision
-        cov = pca.get_covariance()
-        precision = pca.get_precision()
-        assert_array_almost_equal(np.dot(cov, precision),
-                                  np.eye(X.shape[1]), 12)
+    with pytest.raises(ValueError, match=err_msg):
+        pca_fitted.fit(X)
 
-    # test explained_variance_ratio_ == 1 with all components
-    pca = PCA(svd_solver='full')
-    pca.fit(X)
-    assert_almost_equal(pca.explained_variance_ratio_.sum(), 1.0, 3)
+    pca = PCA(n_components=n_components, svd_solver=svd_solver)
```

Comment:
I don't think this is necessary if we're just checking that the attrs are unchanged despite not validating?

Reply:
agreed